### PR TITLE
Eng 15612 - Fix "Alter Table Alter Using MigrateTarget" bug

### PR DESCRIPTION
**Pull request by GuaGuaDean**
This PR concludes the DDL feature for **Migrate Table**.
The latest syntax rule will be: You cannot alter the MIGRATE TO TARGET attribute of the table. You also cannot alter any attributes of the table that affect migration. For example, you cannot add, drop, or alter the USING TTL clause if the table is declared with MIGRATE TO TARGET. And if the table has both MIGRATE TO TARGET and USING TTL, you cannot add, drop, or alter the TTL column. However, you can alter the TTL value, batch size and frequency.

### CODE REVIEW

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java [hunk 1]
```diff
@@ -1117,6 +1120,10 @@ private Statement readTimeToLive(Table table, boolean alter) {
             if (colType != Types.SQL_INTEGER && colType != Types.SQL_BIGINT && colType != Types.SQL_TIMESTAMP) {
                 throw unexpectedToken();
             }
+            // At this moment we don't allow alter TTL column of migrate table on the fly
+            if (alter && table.hasMigrationTarget() && !token.tokenString.equals(table.getTTL().ttlColumn.getNameString())) {
+                throw Error.error(ErrorCode.X_42581, "May not alter TTL column");
```
Comment (Chaozz) [inline src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: Do you have a test case on altering the ttl column on a migration table?

File: tests/frontend/org/voltdb/catalog/TestCatalogDiffs.java [hunk 2]
```diff
@@ -768,7 +768,7 @@ public void testAlterTableTTL() throws IOException {
         Catalog catTTlDropped = catalogForJar(testDir + File.separator + "testAlterTableTTL3.jar");
         verifyDiff(catUpdated, catTTlDropped, false, null, true, false, true);
 
-        builder.addLiteralSchema("\nALTER TABLE A USING TTL 20 MINUTES ON COLUMN C2 BATCH_SIZE 10 MAX_FREQUENCY 3 MIGRATION TO TARGET X;");
+        builder.addLiteralSchema("\nALTER TABLE A ALTER USING TTL 20 MINUTES ON COLUMN C2 BATCH_SIZE 10 MAX_FREQUENCY 3 MIGRATION TO TARGET X;");
```
Comment (kaikaizi) [inline tests/frontend/org/voltdb/catalog/TestCatalogDiffs.java]: Where is the neg test for **add a TTL column when there is not one**? See JIRA.

File: tests/frontend/org/voltdb/catalog/TestCatalogDiffs.java [hunk 3]
```diff
@@ -748,17 +748,17 @@ public void testAlterTableTTL() throws IOException {
         Catalog catOriginal = catalogForJar(testDir + File.separator + "testAlterTableTTL1.jar");
 
         // alter TTL
-        builder.addLiteralSchema("\nALTER TABLE A USING TTL 20 MINUTES ON COLUMN C2;");
+        builder.addLiteralSchema("\nALTER TABLE A ALTER USING TTL 20 MINUTES ON COLUMN C2;");
```
Comment (kaikaizi) [inline tests/frontend/org/voltdb/catalog/TestCatalogDiffs.java]: Why should these fail?

Comment (GuaGuaDean) [reply tests/frontend/org/voltdb/catalog/TestCatalogDiffs.java]: It didn't fail, right? It starts with assertTrue ..

File: tests/frontend/org/voltdb/catalog/TestCatalogDiffs.java (hunk 2 above)
Comment (GuaGuaDean) [reply tests/frontend/org/voltdb/catalog/TestCatalogDiffs.java]: added

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java (hunk 1 above)
Comment (GuaGuaDean) [reply src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: added

**Pull request merged**